examples/js/loaders/SVGLoader.js

```diff
@@ -1614,6 +1614,431 @@ THREE.SVGLoader.prototype = Object.assign( Object.create( THREE.Loader.prototype
 
 } );
 
+THREE.SVGLoader.createShapes = function ( shapePath ) {
+
+	// Param shapePath: a shapepath as returned by the parse function of this class
+	// Returns Shape object
+
+	const BIGNUMBER = 999999999;
+
+	const IntersectionLocationType = {
+		ORIGIN: 0,
+		DESTINATION: 1,
+		BETWEEN: 2,
+		LEFT: 3,
+		RIGHT: 4,
+		BEHIND: 5,
+		BEYOND: 6
+	};
+
+	function findEdgeIntersection( edge1, edge2 ) {
+
+		var x1 = edge1[ 0 ].x;
+		var x2 = edge1[ 1 ].x;
+		var x3 = edge2[ 0 ].x;
+		var x4 = edge2[ 1 ].x;
+		var y1 = edge1[ 0 ].y;
+		var y2 = edge1[ 1 ].y;
+		var y3 = edge2[ 0 ].y;
+		var y4 = edge2[ 1 ].y;
+		var nom1 = ( x4 - x3 ) * ( y1 - y3 ) - ( y4 - y3 ) * ( x1 - x3 );
+		var nom2 = ( x2 - x1 ) * ( y1 - y3 ) - ( y2 - y1 ) * ( x1 - x3 );
+		var denom = ( y4 - y3 ) * ( x2 - x1 ) - ( x4 - x3 ) * ( y2 - y1 );
+		var t1 = nom1 / denom;
+		var t2 = nom2 / denom;
+		var interPoints = [];
+
+		if ( ( ( denom === 0 ) && ( nom1 !== 0 ) ) || ( t1 <= 0 ) || ( t1 >= 1 ) || ( t2 < 0 ) || ( t2 > 1 ) ) {
+
+			//1. lines are parallel or edges don't intersect
+
+			return interPoints;
+
+		} else if ( ( nom1 === 0 ) && ( denom === 0 ) ) {
+
+			//2. lines are colinear
+
+			//check if endpoints of edge2 lies on edge1
+			for ( var i = 0; i < 2; i ++ ) {
+
+				var classify = classifyPoint( edge2[ i ], edge1 );
+				//find position of this endpoints relatively to edge1
+				if ( classify.loc == IntersectionLocationType.ORIGIN ) {
+
+					interPoints.push( { x: edge2[ i ].x, y: edge2[ i ].y, t: classify.t } );
+
+				} else if ( classify.loc == IntersectionLocationType.BETWEEN ) {
+
+					x = + ( ( x1 + classify.t * ( x2 - x1 ) ).toPrecision( 10 ) );
+					y = + ( ( y1 + classify.t * ( y2 - y1 ) ).toPrecision( 10 ) );
```

Comment:
It seems the variables `x` and `y` are not declared before use.

Reply:
True. Thanks, fixed that!